### PR TITLE
feat: implement automatic chain switching & WC v2 chains option

### PR DESCRIPTION
**Pull request by micwallace**

### CODE REVIEW

File: art-gallery-medium-article-website/config-overrides.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = function override(config) {
   config.resolve.fallback = fallback;
   config.plugins = (config.plugins || []).concat([
     new webpack.ProvidePlugin({
-      process: "process/browser",
+      process: "process/browser.js",
```
Comment (nicktaras) [inline art-gallery-medium-article-website/config-overrides.js:16]: @micwallace is this intentional to add .js to this? I think this is a typo.

File: ecommerce-store-website/src/base/utils/interact.js [hunk 2]
```diff
@@ -32,6 +32,12 @@ export const safeMint = async ({
 
 		window.negotiator.ui.dismissLoader()
 
+		// Automatically refresh tokens after 30 seconds
```
Comment (nicktaras) [inline ecommerce-store-website/src/base/utils/interact.js:35]: Only a small thing. Can we remove the comment here. Or change it to 20 seconds

File: ecommerce-store-website/src/providers/TokenContextProvider.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React, { createContext, useState, useEffect } from "react";
+import React, {createContext, useState, useEffect, useMemo} from "react";
```
Comment (nicktaras) [inline ecommerce-store-website/src/providers/TokenContextProvider.js:1]: Awesome! Glad we've added memo.

File: art-gallery-medium-article-website/config-overrides.js (hunk 1 above)
Comment (micwallace) [reply art-gallery-medium-article-website/config-overrides.js:16]: Yeah it's fixed the issue with Magic and probably a few other es modules that use .mjs extension. Similar issue here: 
https://github.com/graphql/graphql-js/issues/2721

File: ecommerce-store-website/src/providers/TokenContextProvider.js (hunk 3 above)
Comment (micwallace) [reply ecommerce-store-website/src/providers/TokenContextProvider.js:1]: Sonar was complaining about it.

File: ecommerce-store-website/src/base/utils/interact.js (hunk 2 above)
Comment (micwallace) [reply ecommerce-store-website/src/base/utils/interact.js]: @nicktaras good catch, done. I was tossing up whether 20 seconds would be enough for discovery providers to index. It should be in most cases.

**Review by nicktaras — APPROVED** [review]

**Review by snowwhitedev — APPROVED** [review]

**Pull request merged**